ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineContext.java

```diff
@@ -238,10 +323,18 @@ void registerQuery(final QueryMetadata query) {
       }
 
       persistentQueries.put(queryId, persistentQuery);
-      metaStore.updateForPersistentQuery(
-          queryId.toString(),
-          persistentQuery.getSourceNames(),
-          ImmutableSet.of(persistentQuery.getSinkName()));
+      if (createAsQuery) {
+        createAsQueries.put(persistentQuery.getSinkName(), queryId);
```

Comment:
why don't we add this query to the `otherQueries`? It can still read from another (or more) source, right? If we don't do that, our error message that says "queries that read from this source: [foo, bar]" will be wrong (can we add a test for that too?)